File: objectModel/Python/cdm/persistence/cdmfolder/attribute_reference_persistence.py
from typing import Optional

from cdm.enums import CdmObjectType
from cdm.objectmodel import CdmAttributeReference, CdmCorpusContext

from .cdm_object_ref_persistence import CdmObjectRefPersistence


class AttributeReferencePersistence(CdmObjectRefPersistence):
    @staticmethod
    def from_data(ctx: CdmCorpusContext, data: str) -> Optional[CdmAttributeReference]:
        if not data:
            return None

        simple_reference = True
        attribute = data

        return ctx.corpus.make_ref(CdmObjectType.ATTRIBUTE_REF, attribute, simple_reference)

File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/persistence/cdmfolder/AttributeReferencePersistence.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder;

import com.fasterxml.jackson.databind.JsonNode;
import com.microsoft.commondatamodel.objectmodel.cdm.CdmAttributeReference;
import com.microsoft.commondatamodel.objectmodel.cdm.CdmCorpusContext;
import com.microsoft.commondatamodel.objectmodel.enums.CdmObjectType;
import com.microsoft.commondatamodel.objectmodel.utilities.CopyOptions;
import com.microsoft.commondatamodel.objectmodel.utilities.ResolveOptions;

public class AttributeReferencePersistence {
  public static CdmAttributeReference fromData(final CdmCorpusContext ctx, final JsonNode obj) {
    final boolean simpleReference = true;
    final String attribute = obj.asText();

    return ctx.getCorpus().makeRef(CdmObjectType.AttributeRef, attribute, simpleReference);
  }

  public static Object toData(final CdmAttributeReference instance, final ResolveOptions resOpt, final CopyOptions options) {
    return CdmObjectRefPersistence.toData(instance, resOpt, options);
  }
}
